libvast/vast/system/partition.hpp

```diff
@@ -83,6 +83,14 @@ struct active_partition_state {
   /// meta index.
   meta_index meta_idx;
 
+  // When creating an addr synopsis for a new column, it is sized for the
+  // absolute worst case: All events in this partition contain a unique ip
+  // address. This wastes a lot of space per partition, so we store the set of
+  // unique ip addresses per column explicitly so we can resize bloom filters to
+  // the minimal possible size when converting this partition to a passive
+  // partition.
+  std::map<qualified_record_field, std::vector<vast::address>> address_fields;
```

Comment:
I'm a bit confused why this state is added to the active partition. We have parallel copy of the partition synopsis in the `meta_index` and that copy is the one that stays in memory.

Did you explore the idea of extending the `synopsis` abstraction with a `finsh()` function and implementing this feature by rewriting the `address_synopsis` itself?

Reply:
Resolved this in the follow-up revisions.